File: modules/native-example/android/src/main/java/com/starter/nativeexample/NativeExampleModule.kt
package com.starter.nativeexample

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeExampleModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NativeExample")

    Constants(
      "pi" to Math.PI
    )

    Function("hello") { name: String ->
      "Hello $name"
    }
  }
}

File: modules/native-example/ios/NativeExampleModule.swift
import ExpoModulesCore

public class NativeExampleModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeExample")

    Constants([
      "pi": Double.pi
    ])

    Function("hello") { (name: String) in
      return "Hello \(name)"
    }
  }
}
